stingray/powerspectrum.py

```diff
@@ -399,3 +399,77 @@ def _make_segment_spectrum(self, lc, segment_size):
             nphots_all.append(np.sum(lc_seg.counts))
 
         return power_all, nphots_all
+
+
+class DynamicalPowerspectrum(AveragedPowerspectrum):
+    def __init__(self, lc, segment_size, norm="frac", gti=None):
+        """
+        Parameters
+        ----------
+        lc : lightcurve.Lightcurve object
+            The time series of which the Dynamical powerspectrum is
+            to be calculated.
+
+        segment_size : float, default 1
+             Length of the segment of light curve, default value is 1 second.
+
+        norm: {"leahy" | "frac" | "abs" | "none" }, optional, default "frac"
+            The normaliation of the periodogram to be used. Options are
+            "leahy", "frac", "abs" and "none", default is "frac".
+
+        Attributes
+        ----------
+        dyn_ps : np.ndarray
+            The matrix with fourier frequencies in the first column and time
+            in the second column.
+        """
+        if segment_size < 2*lc.dt:
+            raise ValueError("Length of the segment is too short to form a "
+                             "light curve!")
+        elif segment_size > lc.tseg:
+            raise ValueError("Length of the segment is too long to create "
+                             "any segments of the light curve!")
+        self.segment_size = segment_size
+        self.norm = norm
+        self.gti = gti
+        self.power_all, _ = AveragedPowerspectrum._make_segment_spectrum(
+            self, lc, segment_size)
+        self._make_matrix()
+
+    def _make_matrix(self):
+        """Create the matrix with freq and time columns."""
+        row1 = self.power_all[0].freq
+        row2 = np.arange(0, self.segment_size*len(row1), self.segment_size)
```

Comment:
Hey @evandromr, we just realized you're not on our slack channel, where we sometimes also discuss stuff. I've just sent you an invite, in case you'd like to join!